blockstore/src/indexed_db_blockstore.rs

```diff
@@ -0,0 +1,160 @@
+use cid::CidGeneric;
+use js_sys::Uint8Array;
+use rexie::{KeyRange, ObjectStore, Rexie, Store, TransactionMode};
+use send_wrapper::SendWrapper;
+use wasm_bindgen::{JsCast, JsValue};
+
+use crate::{convert_cid, Blockstore, BlockstoreError, Result};
+
+/// indexeddb version, needs to be incremented on every schema change
+const DB_VERSION: u32 = 1;
+
+const BLOCK_STORE: &str = "BLOCKSTORE.BLOCKS";
+
+/// A [`Blockstore`] implementation backed by an `IndexedDb` database.
+#[derive(Debug)]
+pub struct IndexedDbBlockstore<const MAX_MULTIHASH_SIZE: usize> {
+    db: SendWrapper<Rexie>,
+}
+
+impl<const MAX_MULTIHASH_SIZE: usize> IndexedDbBlockstore<MAX_MULTIHASH_SIZE> {
+    /// Create or open a [`IndexedDbBlockstore`] with a given name.
+    ///
+    /// # Example
+    /// ```
+    /// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// use blockstore::IndexedDbBlockstore;
+    ///
+    /// let blockstore = IndexedDbBlockstore::<64>::new("blocks").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn new(name: &str) -> Result<Self> {
+        let rexie = Rexie::builder(name)
+            .version(DB_VERSION)
+            .add_object_store(ObjectStore::new(BLOCK_STORE).auto_increment(false))
+            .build()
+            .await
+            .map_err(|e| BlockstoreError::BackingStoreError(e.to_string()))?;
+
+        Ok(Self {
+            db: SendWrapper::new(rexie),
+        })
+    }
+
+    async fn get<const S: usize>(&self, cid: &CidGeneric<S>) -> Result<Option<Vec<u8>>> {
+        let cid = convert_cid::<S, MAX_MULTIHASH_SIZE>(cid)?;
+        let cid = Uint8Array::from(cid.to_bytes().as_ref());
```

Comment:
Do you use the `convert_cid` as a validation mechanism? Because otherwise it is not really needed.

Reply:
removed the MAX_MULTIHASH_SIZE from sled and indexeddb blockstores